src/Type/Comment/CommentType.php

```diff
@@ -81,7 +81,18 @@ private static function fields() {
 						'type' => Types::user(),
 						'description' => __( 'The post field for comments matches the post id the comment is assigned to. This field is equivalent to WP_Comment->comment_post_ID and the value matching the `comment_post_ID` column in SQL.', 'wp-graphql' ),
 						'resolve' => function( \WP_Comment $comment, $args, AppContext $context, ResolveInfo $info ) {
-							$author = new \WP_User( $comment->user_id );
+							if ( ! empty( $comment->user_id ) ) {
+								$author = new \WP_User( $comment->user_id );
+							} else {
+								$author = new \WP_User();
+
+								if ( ! empty( $comment->comment_author ) ) {
+									$author->__set( 'display_name', $comment->comment_author );
+								}
+								if ( ! empty( $comment->comment_author_email ) ) {
+									$author->__set( 'user_email', $comment->comment_author_email );
+								}
```

Comment:
We need to populate the new WP_User's `user_url ` field, using `$comment->comment_author_url` as that's a field that can be queried against the author and the data to populate it exists here.

Also, we need to populate the ID field to be unique, as the ID is used to normalize data in a client side cache, so without a unique ID, client side cache will be funky. 

Without setting a unique ID, anytime a comment author is queried for, it will have an ID of 0, which means every single author that's queried will bust the cache of the previous author because they have the same ID.

I'm thinking we could use the email address and name as the unique identifier. . . something like: 

```
$comment_author_id = [ 'type' => 'comment_author', 'email' => $comment->comment_author_email, 'name' => $comment->comment_author ];
$author->_set( 'ID', $comment_author_id );
```

This will translate to a base64 encoded ID that has enough info to query for that author again, but will also ensure that it's unique enough that client side caches don't accidentally override other comment authors in the cache. 

Then of course this will need to be tested as well.

Reply:
@pjpak did you see this comment too?

Reply:
I added some changes for comment_author_id at https://github.com/wp-graphql/wp-graphql/pull/240/files#diff-ee902c2db37a04884a3f7a15426e2b9eR94
I wasnt quite sure how to test that.